webpush/push.go

```diff
@@ -69,6 +69,25 @@ func NewPushRequest(sub *Subscription, message string, token string) (*http.Requ
 	return req, nil
 }
 
+// NewVapidRequest creates a valid Web Push HTTP request for sending a message
+// to a subscriber, using Vapid authentication. You can add more headers to
+// configure collapsing, TTL.
+func NewVapidRequest(sub *Subscription, message string, vapid *Vapid) (*http.Request, error) {
+	req, err := NewPushRequest(sub, message, "")
+	if err != nil {
+		return nil, err
+	}
+
+	tok, err := vapid.Token(sub.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Authorization", fmt.Sprintf(`Bearer %s`, tok))
```

Comment:
The [latest spec](https://tools.ietf.org/html/draft-ietf-webpush-vapid-01) switched from `Bearer` to `WebPush`. According to my test, this is now supported by both FCM's and Mozilla's servers.